api/searchCases/index.ts

```diff
@@ -9,22 +9,25 @@ import { ElasticSearchQuery } from './interfaces/ElasticSearchQuery';
  * Manually creating Elastic search query
  */
 export function modifyRequest(proxyReq, req) {
-  const userInfo = req.session.passport.user.userinfo as UserInfo;
-  const request = prepareElasticQuery(req.query, req.body, userInfo);
+  const userInfo = getUserInfoFromRequest(req);
 
-  // Write out body changes to the proxyReq stream
-  const body = JSON.stringify(request);
+  if (userInfo) {
+    const request = prepareElasticQuery(req.query, req.body, userInfo);
 
-  // Update header
-  proxyReq.setHeader('content-type', 'application/json');
-  proxyReq.setHeader('content-length', body.length);
+    // Write out body changes to the proxyReq stream
+    const body = JSON.stringify(request);
 
-  // Write out body changes to the proxyReq stream
-  proxyReq.write(body);
+    // Update header
+    proxyReq.setHeader('content-type', 'application/json');
+    proxyReq.setHeader('content-length', body.length);
 
-  // Remove body-parser body object from the request
-  delete req.body;
-  proxyReq.end();
+    // Write out body changes to the proxyReq stream
+    proxyReq.write(body);
+
+    // Remove body-parser body object from the request
+    delete req.body;
```

Comment:
Both these statements need to be out of the block

Reply:
Should we amend the request even when it is invalid?

Reply:
until unless proxy.end is called it will not work.
delete req.body;
```suggestion
    delete req.body;
```
proxyReq.end();